### ISSUE TITLE
High RAM usage due to LeechBlockNG.

### ISSUE THREAD

**Issue by pras92** [proginosko/LeechBlockNG#42]
Hello,
Firstly, thanks a lot proginosko, for continuing development with Quantum. 

It's the 1st time I'm using the 'block with keyword' function and using * with ~ takes the FF ram usage from about 600MB to 3800MB. 

I just now posted some details about this here, https://www.reddit.com/r/firefox/comments/7tpjru/firefox_quantum_high_ram_usage/du8ikt7/

Thanks.

**Comment by proginosko** [comment]
Interesting. I'm not finding the same problem on my setup (FF 58.0.2 64-bit on Win10).

Can you give a bit more info on how the RAM usage rises? Does it rise progressively over time without the user doing anything (navigating pages, opening new tabs)? If not, what events cause it to jump upwards?

**Comment by Fickri-s** [comment]
I'm having similar issue. RAM usage increases progressively over time, with sharp increase when the page is opened.

**Comment by pras92** [comment]
Hello,
Sorry for such a delayed response. I'm still sorting out my inbox. 

When facing the issue, I too had FF 58.0.2, now FF59.0 on Win10 64bit. 

To answer your questions, yes it did rise progressively over time without any user actions. The RAM rose to peak in about 15 seconds. Reversing the LB settings did reduce the RAM usage slowly. 

I did some more testing right now, and found something else. 

-> It seems, (in addition to *) using just a keyword without ~ is what causes the ultra high RAM spike in a very short time, along with system lagging. 
-> (In addition to *), using a keyword with ~ causes slightly higher-than-normal RAM usage slowly but steadily, however, the system lag is just the same as in the previous case.

(Prior to my original post, I had a combination of keywords with and without ~ and hence couldn't figure this out previously.)

By system lag, I mean about 3 to 10 seconds of lag in both in-browser and other operations.

I do use a couple other highly trusted extensions, though these results are with other extensions being disabled. Mine is a new laptop with 8GB RAM and FFQ running in SSD. 

Thank you for your time.

**Comment by Erudition** [comment]
Same issue here. LeechBlock causes the Firefox process RAM usage to climb by 1 GB every 30 seconds, making the browser unusable and the whole system laggy. Disabling Leechblock brings the browsers RAM back down under a gig and performance is back.

**Comment by proginosko** [comment]
> Same issue here. LeechBlock causes the Firefox process RAM usage to climb by 1 GB every 30 seconds, making the browser unusable and the whole system laggy. Disabling Leechblock brings the browsers RAM back down under a gig and performance is back.

Do you have a long list of sites to block? Which OS are you using?

Please export your options to a text file (Options > General > Export Options) and email it to me at leechblockng@proginosko.com so I can take a closer look.

**Comment by Erudition** [comment]
> Do you have a long list of sites to block? Which OS are you using?

No, I just installed it and only blocked `*.youtube.com` witha single Block Set, switched to the Delaying page. All other settings are default. Using Firefox 128 (Nightly) on Gnu/Linux.
 
> Please export your options to a text file (Options > General > Export Options) so I can take a closer look.

Okay, I'll reinstall and try to reproduce the issue.

**Comment by Drulac** [comment]
Hello :grin: 
I have the same issue on 128.5.0esr (64 bits) with debian (linux). The issue was present with previous versions of firefox.
I have only 4 sites blocked. I've set a very? long (10 years) blocking time, maybe it come from that ? I will try

Have a good day !
